### PR TITLE
Add tests for core and NLU

### PR DESCRIPTION
**Pull request by MetcalfeTom**
To solve [this issue](https://github.com/RasaHQ/rasa_core/issues/1427)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - /tmp/cached/
+python:
+- '3.5'
+- '3.6'
+install:
+- pip install -r requirements.txt
+- pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_md-2.0.0/en_core_web_md-2.0.0.tar.gz --no-cache-dir > jnk
+- python -m spacy link en_core_web_md en
+script:
+- py.test test_stack.py
```
Comment (wochinge) [inline .travis.yml]: add a newline here

File: test_stack.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    trainer = Trainer(config.load("nlu_config.yml"))
+    interpreter = trainer.train(training_data)
+    test_interpreter_dir = trainer.persist("./models/nlu", fixed_model_name="test")
+
+    assert interpreter.parse('hello') is not None
+    assert test_interpreter_dir
+
+
+def test_agent_and_persist():
+    agent = Agent('domain.yml', policies=[MemoizationPolicy(), KerasPolicy(epochs=2), FallbackPolicy()])
```
Comment (wochinge) [inline test_stack.py]: How about using the policy configuration `policies.yml`? So we actually test whether our provided configuration works?

File: test_stack.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import uuid
+
+
+def test_nlu_interpreter():
+    training_data = load_data("data/nlu_data.md")
+    trainer = Trainer(config.load("nlu_config.yml"))
+    interpreter = trainer.train(training_data)
+    test_interpreter_dir = trainer.persist("./models/nlu", fixed_model_name="test")
+
+    assert interpreter.parse('hello') is not None
```
Comment (wochinge) [inline test_stack.py]: why not checking for the actual intent?

File: requirements.txt [hunk 4]
```diff
@@ -20,16 +20,16 @@ coloredlogs==10.0
 flask==1.0.2
 flask_cors==3.0.4
 scikit-learn==0.19.1
-rasa_nlu~=0.13.0
+rasa_nlu
```
Comment (wochinge) [inline requirements.txt]: Mhm, I am not sure about removing the version annotations. 
Pro:
- it tests the latest version on pypi (!= master version)

Contra:
- the other dependencies might not equal the ones we are using in rasa_core or rasa_nlu (e.g. if their version number increased with the core/nlu version) and we would need a pr anyway.

File: .travis.yml (hunk 1 above)
Comment (wochinge) [inline .travis.yml]: py.test is not in the dependencies right? We need a `dev-requirements.txt`  then.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+language: python
+sudo: enabled
+cache:
+  directories:
+  - $HOME/.cache/pip
+  - /tmp/cached/
+python:
+- '3.5'
+- '3.6'
```
Comment (wochinge) [inline .travis.yml]: Should we add python2? If not, should we add a note in the readme that it's not tested with python2?

Comment (MetcalfeTom) [reply .travis.yml]: We're dropping support for python2 shortly so I thought it was not worth adding.

File: test_stack.py (hunk 2 above)
Comment (MetcalfeTom) [reply test_stack.py]: Like it.  I've added the `config.load` line but for the sake of training times I still want to alter the number of epochs for the `KerasPolicy`.

File: requirements.txt (hunk 4 above)
Comment (wochinge) [reply requirements.txt]: @tmbo What do you think about this?

File: .travis.yml (hunk 5 above)
Comment (wochinge) [reply .travis.yml]: 👍

File: requirements.txt (hunk 4 above)
Comment (MetcalfeTom) [reply requirements.txt]: I guess the better option is to keep this annotation and then clone the master branch in the Travis script.

**Comment by MetcalfeTom** [comment]
- Decided against testing the makefile, since we are not likely to make dramatic changes to the way we parse arguments from the command line.  It would also involve training another model which would increase build time.
- Since this starter pack is made for newcomers, we would like to keep obfuscation low and a `dev-requirements.txt` would hinder this, so no `pycodestyle` testing either

**Pull request merged**